src/test/java/com/rakovets/course/java/core/practice/string/StringUtilTest.java

```diff
@@ -0,0 +1,289 @@
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments> identicalStrings() {
+        return Stream.of(
+                Arguments.of("true", "Hello", "Hello"),
+                Arguments.of("false", "Hello", "hello"),
+                Arguments.of("true", " ", " ")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("identicalStrings")
+    void identicalStrings(boolean expectedSymbol, String str1, String str2) {
+        // GIVEN
+
+        // WHEN
+        boolean actualString = StringUtil.identicalStrings(str1, str2);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments> removeSpacesAroundCastUpperCase() {
+        return Stream.of(
+                Arguments.of("HELLO", "Hello "),
+                Arguments.of("HELLO", " HeLlo"),
+                Arguments.of("", " ")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("removeSpacesAroundCastUpperCase")
+    void removeSpacesAroundCastUpperCase(String expectedSymbol, String str1) {
+        // GIVEN
+
+        // WHEN
+        String actualString = StringUtil.removeSpacesAroundCastUpperCase(str1);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments> extractionSubstring() {
+        return Stream.of(
+                Arguments.of("ll", "Hello  ", 2, 4),
+                Arguments.of("l", " HeLlo", 4, 5),
+                Arguments.of("lati", "Relationship", 2, 6)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("extractionSubstring")
+    void extractionSubstring(String expectedSymbol, String str1, int nSymbol, int mSymbol) {
+        // GIVEN
+
+        // WHEN
+        String actualString = StringUtil.extractionSubstring(str1, nSymbol, mSymbol);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>happyEmoji() {
+        return Stream.of(
+                Arguments.of("Hello:)  ", "Hello:(  ", "(", ")"),
+                Arguments.of(" HeLlo)", " HeLlo(", "(", ")"),
+                Arguments.of(":):):)", ":(:(:(", "(", ")")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("happyEmoji")
+    void happyEmoji(String expectedSymbol, String str1, String previousSymbol, String modernSymbol) {
+        // GIVEN
+
+        // WHEN
+        String actualString = StringUtil.happyEmoji(str1, previousSymbol, modernSymbol);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>startEndWith() {
+        return Stream.of(
+                Arguments.of("true", "Hi", "Hi"),
+                Arguments.of("false", "hi !", " Java"),
+                Arguments.of("false", "!Hi", "java")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("startEndWith")
+    void startEndWith(boolean expectedSymbol, String text, String word) {
```

Comment:
"Violation: 'nameMethodTest' was not used for test-method naming
Solution: Rename method
"